java/vector/src/main/java/org/apache/arrow/vector/complex/StructVector.java

```diff
@@ -482,6 +482,9 @@ public Object getObject(int index) {
   @Override
   public void get(int index, ComplexHolder holder) {
     holder.isSet = isSet(index);
+    if (holder.isSet == 0) {
+      return;
```

Comment:
should we null out reader here? Is this actually a bug? it seems different then #4543  since i doesn't look like an exception will be thrown by calling super.get below?

Reply:
Good suggestion. It should be nulled out.

As you have observed, it is different from #4543 as no exception is thrown. However, it is just useless to retrieve the reader by calling super.get.

Reply:
Agreed, I just don't know if there are some (bad) assumptions being made downstream by code someplace.  CC @praveenbingo  @pravindra  if you don't have any objections please go ahead and merge (or I will merge tomorrow if I don't here any)